Xamarin.Forms.Build.Tasks/MethodDefinitionExtensions.cs

```diff
@@ -28,7 +28,7 @@ public static bool MatchXArguments(this MethodDefinition methodDefinition, Eleme
 			for (var i = 0; i < methodDefinition.Parameters.Count; i++)
 			{
 				var paramType = methodDefinition.Parameters[i].ParameterType;
-				var argType = ((IElementNode)arguments[i]).XmlType.GetTypeReference(module, null);
+				var argType = context.Variables [arguments [i] as IElementNode].VariableType;
```

Comment:
Instead of matching x:Arguments on the XmlType of the node, I'm matching them on the VariableType of the variable holding the actual value of the node.